scalameta/io/js/src/main/scala/scala/meta/io/PlatformIO.scala

```diff
@@ -0,0 +1,15 @@
+package scala.meta.io
+
+import java.io.InputStream
+import java.nio.charset.Charset
+
+object PlatformIO {
+  def workingDirectory: AbsolutePath = AbsolutePath(fileSeparator)
+  def read(stream: InputStream, charset: Charset): String =
+    throw new IllegalStateException("Reading from java.io.InputStream is not supported in JavaScript.")
```

Comment:
You don't need Node.js libraries for this. You can implement it in portable Scala.js code. We support both `InputStream` and `Charset`, as well as `InputStreamReader`. You simply have to avoid `scala.io`, but everyone should do that anyway.

Wrap the `stream` in a `new InputStreamReader(stream, charset)`, then loop on its `read` method until you exhaust the input, while appending everything in the result string. Something like the following (which is even portable on the JVM too, and faster than `scala.io.Source`):
```scala
def read(stream: InputStream, charset: Charset): String = {
  val reader = new InputStreamReader(stream, charset)
  val buffer = new Array[Char](4096)
  val builder = new java.lang.StringBuilder

  @tailrec def loop(): Unit = {
    val charsRead = reader.read(buffer)
    if (charsRead >= 0) {
      builder.append(buffer, 0, charsRead)
      loop()
    }
  }

  loop()
  reader.close() // check whether you need this
  builder.toString()
}
```